_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+
+```shell
+CONTEXT=$(kubectl config current-context)
+
+hal config provider kubernetes account add spinnaker-gke-account \
+    --context $CONTEXT
+```
+
+### Choose the installation environment
+
+The __Distributed installation__ on Kubernetes deploys each Spinnaker's microservice separately to a remote cloud, and it is the recommended environment for zero-downtime updates of Spinnaker.
+
+Select the distributed deployment with:
+
+```shell
+hal config deploy edit --type distributed --account-name spinnaker-gke-account
+```
+
+### Choose a storage service
+
+Spinnaker requires an external storage provider for persisting application settings and configured pipelines. To avoid losing this data, using a hosted storage solution is recommended. For this example, use Google Cloud Storage. Hosting the data externally allows to delete clusters in between sessions, and keep the pipelines once the clusters have been recreated.
+
+In the `spinnaker` folder, run the following lines:
+
+```shell
+SERVICE_ACCOUNT_NAME=spinnaker-gcs-account
+SERVICE_ACCOUNT_DEST=~/.gcp/gcs-account.json
+
+gcloud iam service-accounts create \
+    $SERVICE_ACCOUNT_NAME \
+    --display-name $SERVICE_ACCOUNT_NAME
+
+SA_EMAIL=$(gcloud iam service-accounts list \
+    --filter="displayName:$SERVICE_ACCOUNT_NAME" \
+    --format='value(email)')
+
+PROJECT=$(gcloud config get-value project)
+
+gcloud projects add-iam-policy-binding $PROJECT \
+    --role roles/storage.admin --member serviceAccount:$SA_EMAIL
+
+mkdir -p $(dirname $SERVICE_ACCOUNT_DEST)
+
+gcloud iam service-accounts keys create $SERVICE_ACCOUNT_DEST \
+    --iam-account $SA_EMAIL
+
+BUCKET_LOCATION=us
+
+hal config storage gcs edit --project $PROJECT \
+    --bucket-location $BUCKET_LOCATION \
+    --json-path $SERVICE_ACCOUNT_DEST
+
+hal config storage edit --type gcs    
+```
+
+The last `hal` commands edit the storage settings and set the storage source to GCS.
+
+### Deploy Spinnaker
+
+You can verify the deployment configuration with:
+
+```shell
+hal config deploy
+```
+
+And the storage configuration with:
+```shell
+hal config storage
+```
+
+List the available Spinnaker versions:
+
+```shell
+hal version list
+```
+
+Set the version for the deployment with:
+
+```shell
+hal config version edit --version $VERSION
+```
+
+Deploy Spinnaker to GKE:
+
+```shell
+hal deploy apply
+```
+
+Even if `hal deploy apply` returns successfully, the installation may not be complete yet. Check if the pods are ready with `k9s`:
+
+```shell
+k9s -n spinnaker
+```
+
+A healthy Spinnaker deployment will look like this:
+
+{% img blog/jhipster-ci-cd/spinnaker-k9s.png alt:"Spinnaker k9s view" width:"800" %}{: .center-image }
+
+Then connect to the Spinnaker UI with:
+
+```shell
+hal deploy connect
+```
+Navigate to [localhost:9000](localhost:9000), the UI will look like this:
+
+{% img blog/jhipster-ci-cd/spinnaker-ui.png alt:"Spinnaker DeckUI" width:"800" %}{: .center-image }
```

Comment:
I see the top part, but not the bottom part with pagination.

<img width="1587" alt="Screen Shot 2022-08-09 at 00 12 06" src="https://user-images.githubusercontent.com/17892/183577399-9ff1e4d8-1993-4569-b4e1-6583d99a68d6.png">